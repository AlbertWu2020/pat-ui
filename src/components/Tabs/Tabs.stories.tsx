import React from 'react';
import { action } from '@storybook/addon-actions';
import Tabs from '../Tabs/Tabs';

export default {
  title: 'Tabs',
  component: Tabs,
};

export const DefaultTabs = () => {
  return (
    <div onChange={action('defaultTab')}>
      <Tabs
        tabs={[
          {
            heading: 'Tab1',
            content: <div>This is tab1</div>,
          },
          {
            heading: 'Tab2',
            content: <div>This is tab2</div>,
            active: true,
          },
          {
            heading: 'Tab3',
            content: <div>This is tab3</div>,
          },
        ]}
      ></Tabs>
    </div>
  );
};
