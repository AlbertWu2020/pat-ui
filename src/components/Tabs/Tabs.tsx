import React from 'react';

interface IProps {
  headings: string[];
}
interface IState {
  activeHeading: string;
  description: string[];
  finalDescription: string;
}

class Tabs extends React.Component<IProps, IState> {
  public constructor(props: IProps) {
    super(props);
    this.state = {
      activeHeading:
        this.props.headings && this.props.headings.length > 0
          ? this.props.headings[0]
          : '',
      description: ['This is Tab1', 'This is Tab2'],
      finalDescription: 'This is Tab1',
    };
  }

  handleTabClick = (e: React.MouseEvent<HTMLLIElement>) => {
    const li = e.target as HTMLLIElement;
    const heading: string = li.textContent ? li.textContent : '';

    if (li.textContent === 'Tab1') {
      this.setState({ finalDescription: this.state.description[0] });
    } else this.setState({ finalDescription: this.state.description[1] });
    this.setState({ activeHeading: heading });
  };

  public render() {
    return (
      <div>
        <ul className='tabs'>
          {this.props.headings.map((heading, index: number) => (
            <li
              key={index}
              onClick={this.handleTabClick}
              className={heading === this.state.activeHeading ? 'active' : ''}
            >
              {heading}
            </li>
          ))}
        </ul>
        <div>{this.state.finalDescription}</div>
      </div>
    );
  }
}

export default Tabs;
