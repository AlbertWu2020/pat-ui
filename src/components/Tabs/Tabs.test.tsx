import React from 'react';

import { render } from '@testing-library/react';

import Tabs from '../Tabs/Tabs';

describe('Tabs', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(
      <Tabs
        tabs={[{ heading: 'Tab1', content: <div>This is tab1</div> }]}
      ></Tabs>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('should render tab1', () => {
    const tabProps = {
      tabs: [
        {
          heading: 'tab1',
          content: <div>content</div>,
        },
      ],
    };
    const wrapper = render(<Tabs {...tabProps}></Tabs>);
    const element = wrapper.queryByText('tab1') as HTMLElement;
    expect(element).toBeInTheDocument();
    expect(element.tagName).toBe('LI');
  });
});
