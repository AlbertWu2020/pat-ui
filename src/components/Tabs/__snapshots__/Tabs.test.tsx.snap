// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <ul
      class="tabs"
    >
      <li
        class="active"
      >
        Tab1
      </li>
    </ul>
    <div>
      <div>
        This is tab1
      </div>
    </div>
  </div>
</DocumentFragment>
`;
